birdyboard.py

```diff
@@ -0,0 +1,505 @@
+                pass
+        else:  # select the number of the printed tweet.
+            try:
+                next_step == int(next_step)
+            except ValueError:
+                print("you didn't enter a number.")
+                self.view_chirps_next_step()
+            finally:
+                # TODO: mess with this until it works.
+                if int(next_step)-1 >= 0:
+                    try:
+                        self.chirp_index = int(next_step)-1
+                        checker = self.chirps_library[self.public_or_private][self.chirp_index]
+                        self.view_full_chirp()
+                        self.full_chirp_menu()
+                    except IndexError:
+                        print("your number is not in the list of chirps.")
+                        self.view_chirps_next_step()
+                    finally:
+                        pass
+                else:
+                    print("your number is not in the list of chirps.")
+                    self.view_chirps_next_step()
+
+# ########################################
+# ######## VIEW FULL CHIRP THREAD ########
+# ########################################
+
+    def view_full_chirp(self):
+        """
+        displays a full chirp based on the index chosen in view_chirps_next_step- public or private depending on top public_or_private variable.
+        Arguments: none
+        """
+        if self.public_or_private == "public":
+            [print(chirp[0] + ": " + chirp[1]) for chirp in self.chirps_library[self.public_or_private][self.chirp_index]]
+        elif self.public_or_private == "private":
+            [print(chirp[0] + ": " + chirp[1]) for chirp in self.chirps_library[self.public_or_private][self.chirp_index]["chirps"]]
+
+    def full_chirp_menu(self):
+        """
+        prints after view_full_chirp to ask whether the user would like to go back, exit, or add to the chirp thread.
+        arguments: none
+        """
+        if self.user_name is not None:
+            next_step = input("1. add a new chirp to this thread.\n'b' to go back.\n'x' to exit.\n>> ")
+        else:
+            next_step = input("'b' to go back.\n'x' to exit.\n>> ")
+
+        if next_step == "b":
+            print("going back.")
+            self.current_chirp = None
+            self.view_chirps()
+            self.view_chirps_next_step()
+        elif next_step == "x":
+            print("goodbye.")
+            exit()
+        elif next_step == "1":
+            if self.user_name is not None:
+                self.add_to_chirp_menu()
+            else:
+                print("command not recognized.")  # to keep an un-logged in user from pressing 1 and finding a SECRET INROAD to chirping.
+                self.chirp_thread_menu()
+        else:
+            print("command not recognized.")
+            self.chirp_thread_menu()
+
+# #############################
+# ######## ADD COMMENT ########
+# #############################
+
```

Comment:
Your "Add comment" and "New chirp thread" are good candidates for rolling into their own class.